subprojects/base-services/src/main/java/org/gradle/internal/classloader/ClasspathUtil.java

```diff
@@ -42,7 +41,7 @@ public static void addUrl(URLClassLoader classLoader, Iterable<URL> classpathEle
             for (URL url : classLoader.getURLs()) {
                 original.add(toURI(url));
             }
-            JavaMethod<URLClassLoader, Object> method = JavaReflectionUtil.method(URLClassLoader.class, Object.class, "addURL", URL.class);
+            JavaMethod<URLClassLoader, Object> method = JavaMethod.method(URLClassLoader.class, Object.class, "addURL", URL.class);
```

Comment:
`method()` sounds redundant, should this be called `of()` or `create()`?

Reply:
There are `method` and `staticMethod` factory methods on `JavaMethod`. I went with those two names to distinguish static and instance methods. Do you have a better suggestion?

Reply:
`create()` (or even `createInstance()`) vs `createStatic()`? Also, `ofInstance()` vs `ofStatic()`?

Reply:
Maybe `fromInstance()` vs `fromStatic()`?

Reply:
I changed the factory method names to `of` and `ofStatic`.